src/sklearn_evaluation/plot/pca.py

```diff
@@ -0,0 +1,150 @@
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from ..telemetry import SKLearnEvaluationLogger
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+@SKLearnEvaluationLogger.log(feature='plot')
+def pca(X, Y=None, n_components=2, standard_scalar=False, ax=None):
+    """
+    Plot principle component analysis curve.
+
+    Parameters
+    ----------
+    
+    X : array-like, shape = [n_samples, n_features]
+        Training data, where n_samples is the number of samples and
+        n_features is the number of features
+    
+    Y : array-like or list or None
+        set None if Ignored otherwise, pass the targets here
+        
+    n_components : int, float or 'mle', default=None
+        Number of components to keep.
+        if n_components is not set all components are kept::
+
+            n_components == min(n_samples, n_features)
+
+        If ``n_components == 'mle'`` and ``svd_solver == 'full'``, Minka's
+        MLE is used to guess the dimension. Use of ``n_components == 'mle'``
+        will interpret ``svd_solver == 'auto'`` as ``svd_solver == 'full'``.
+
+        If ``0 < n_components < 1`` and ``svd_solver == 'full'``, select the
+        number of components such that the amount of variance that needs to be
+        explained is greater than the percentage specified by n_components.
+
+        If ``svd_solver == 'arpack'``, the number of components must be
+        strictly less than the minimum of n_features and n_samples.
+
+        Hence, the None case results in::
+
+            n_components == min(n_samples, n_features) - 1
+    
+    standard_scalar : Flag 
+        set True if features needs to be scaled
+    
+    ax : matplotlib Axes
+        Axes object to draw the plot onto, otherwise uses current Axes
+
+    Notes
+    -----
+    It is assumed that the X parameter are in array format for features
+
+
+    Returns
+    -------
+    ax: matplotlib Axes
+        Axes containing the plot
+
+    Examples
+    --------
+    .. plot:: ../examples/pca.py
+
+    """
+    if np.isnan(X).sum():
+        raise ValueError('X array should not consist nan '
+                         'pca')
+        
+    if ax is None:
+        ax = plt.gca()
+    
+    # Standardizing the features
+    if standard_scalar:
+        X = StandardScaler().fit_transform(X)
+    
+    # PCA Projection to 2D
+    _pca(X, Y, n_components, ax)
+
+    return ax
+
+
+def _pca(X, Y=None, n_components=2, ax=None):
```

Comment:
There's a method named `pca` already, consider renaming this to something more specific